src/pages/home/report/withReportAndPrivateNotesOrNotFound.js

```diff
@@ -0,0 +1,127 @@
+
+    /** Information about the network */
+    network: networkPropTypes.isRequired,
+
+    /** Session of currently logged in user */
+    session: PropTypes.shape({
+        /** Currently logged in user accountID */
+        accountID: PropTypes.number,
+    }),
+
+    route: PropTypes.shape({
+        /** Params from the URL path */
+        params: PropTypes.shape({
+            /** reportID and accountID passed via route: /r/:reportID/notes */
+            reportID: PropTypes.string,
+            accountID: PropTypes.string,
+        }),
+    }).isRequired,
+};
+
+const defaultProps = {
+    forwardedRef: () => {},
+    report: {},
+    session: {
+        accountID: null,
+    },
+};
+
+export default function (WrappedComponent) {
+    // eslint-disable-next-line rulesdir/no-negated-variables
+    function WithReportAndPrivateNotesOrNotFound({forwardedRef, ...props}) {
+        const {route, report, network, session} = props;
+        const accountID = route.params.accountID;
+        const isLoadingPrivateNotes = report.isLoadingPrivateNotes;
+
+        useEffect(() => {
+            if (network.isOffline && report.isLoadingPrivateNotes) {
+                return;
+            }
+
+            Report.getReportPrivateNote(report.reportID);
+            // eslint-disable-next-line react-hooks/exhaustive-deps -- do not add isLoadingPrivateNotes to dependencies
+        }, [report.reportID, network.isOffline]);
+
+        const isPrivateNotesEmpty = accountID ? _.isEmpty(lodashGet(report, ['privateNotes', accountID, 'note'], '')) : _.isEmpty(report.privateNotes);
+        const shouldShowFullScreenLoadingIndicator = isLoadingPrivateNotes !== false && isPrivateNotesEmpty;
+
+        // eslint-disable-next-line rulesdir/no-negated-variables
+        const shouldShowNotFoundPage = useMemo(() => {
+            // Show not found view if the report is archived, or if the note is not of current user.
+            if (ReportUtils.isArchivedRoom(report) || (accountID && Number(session.accountID) !== Number(accountID))) {
+                return true;
+            }
+
+            // Don't show not found view if the notes are still loading, or if the notes are non-empty.
+            if (isLoadingPrivateNotes !== false || !isPrivateNotesEmpty) {
+                return false;
+            }
+
+            // As notes being empty and not loading is a valid case, show not found view only in offline mode.
+            return network.isOffline;
+        }, [report, network.isOffline, accountID, session.accountID, isPrivateNotesEmpty, isLoadingPrivateNotes]);
+
+        if (shouldShowNotFoundPage) {
+            return <NotFoundPage />;
+        }
+
+        if (shouldShowFullScreenLoadingIndicator) {
+            return <FullScreenLoadingIndicator />;
+        }
+
+        return (
+            <WrappedComponent
+                // eslint-disable-next-line react/jsx-props-no-spreading
+                {...props}
+                ref={props.forwardedRef}
```

Comment:
forwardedRef is destructed in line 55, it no longer exists in props.
```suggestion
                ref={forwardedRef}
```

Reply:
my bad. Fixed. thanks.